src/python/pants/backend/docker/lint/hadolint/rules_integration_test.py

```diff
@@ -0,0 +1,150 @@
+# Copyright 2021 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import annotations
+
+from textwrap import dedent
+
+import pytest
+
+from pants.backend.docker.lint.hadolint.rules import HadolintFieldSet, HadolintRequest
+from pants.backend.docker.lint.hadolint.rules import rules as hadolint_rules
+from pants.backend.docker.target_types import DockerImage
+from pants.core.goals.lint import LintResult, LintResults
+from pants.core.util_rules import config_files, external_tool, source_files
+from pants.engine.addresses import Address
+from pants.engine.target import Target
+from pants.testutil.rule_runner import QueryRule, RuleRunner
+
+
+@pytest.fixture
+def rule_runner() -> RuleRunner:
+    return RuleRunner(
+        rules=[
+            *hadolint_rules(),
+            *config_files.rules(),
+            *external_tool.rules(),
+            *source_files.rules(),
+            QueryRule(LintResults, [HadolintRequest]),
+        ],
+        target_types=[DockerImage],
+    )
+
+
+GOOD_FILE = dedent(
+    """
+    FROM python:3.8
+    """
+)
+
+BAD_FILE = dedent(
+    """
+    FROM python
+    """
+)
+
+
+def run_hadolint(
+    rule_runner: RuleRunner, targets: list[Target], *, extra_args: list[str] | None = None
+) -> tuple[LintResult, ...]:
+    rule_runner.set_options(
+        ["--backend-packages=pants.backend.docker.lint.hadolint", *(extra_args or ())],
+        env_inherit={"PATH"},
```

Comment:
Does this end up being necessary? I suspect it isn't.

Reply:
Ah, dunno. Can try without ;)

Reply:
Goner :)